eng/common/TestResources/New-TestResources.ps1

```diff
@@ -289,7 +305,7 @@ if ($CI) {
     # Set the resource group name variable.
     Write-Host "Setting variable 'AZURE_RESOURCEGROUP_NAME': $ResourceGroupName"
     Write-Host "##vso[task.setvariable variable=AZURE_RESOURCEGROUP_NAME;]$ResourceGroupName"
-    $environmentVariables['AZURE_RESOURCEGROUP_NAME'] = $ResourceGroupName
+    $EnvironmentVariables['AZURE_RESOURCEGROUP_NAME'] = $ResourceGroupName
```

Comment:
This could also potentially overwrite although not likely.

Reply:
If we end up with more cases maybe we split these into another env variable list and then use the same merge function you wrote for the others.